src/number_theory/number_field.lean

```diff
@@ -249,4 +251,113 @@ end
 
 end roots
 
+section bounded
+
+open finite_dimensional polynomial set
+
+variables (K : Type*) [field K] [number_field K]
+variables (A : Type*) [normed_field A] [is_alg_closed A] [normed_algebra ℚ A]
+
+lemma finite_of_norm_le (B : ℝ) :
+  {x : K | is_integral ℤ x ∧ ∀ φ : K →+* A, ∥ φ x ∥  ≤ B}.finite :=
```

Comment:
```suggestion
  {x : K | is_integral ℤ x ∧ ∀ φ : K →+* A, ∥ φ x ∥ ≤ B}.finite :=
```
This is a pretty big proof, do you think it is possible to extract one or more sublemmas?

Reply:
Well, there are basically two parts: 1) show that the coefficients of the minpoly are uniformly bounded, 2) show that there are only finitely many such polynomials and thus finitely many roots. It used to be two different lemmas but I merged them into one since there was some (small) overlaps. I can split it again if you want. It would something like that:
```lean
example {x : K} (B : ℝ) (h : ∀ φ : K →+* A, ∥φ x∥ ≤ B) :
  ∃ C : ℕ, ∀ i, |(minpoly ℚ x).coeff i| ≤ C := by sorry
```

Reply:
Yes please, it's better to have two lemmas even if there are some overlaps. Thanks!

I am investigating the `char_zero` problem above.